Sources/SourceControl/RepositoryManager.swift

```diff
@@ -267,10 +280,69 @@ public class RepositoryManager {
                     // Fetch the repo.
                     var fetchError: Swift.Error? = nil
                     do {
-                        // Start fetching.
-                        try self.provider.fetch(repository: handle.repository, to: repositoryPath)
+                        let cache = try await { self.cacheManager?.lookup(repository: handle.repository, completion: $0) }
+
+                        try self.provider.fetch(repository: cache.repository, to: repositoryPath)
```

Comment:
I think we should look into whether a straight up copy might be faster than performing a fetch here.

Reply:
That's a good idea, and should also eliminate the need to reset the remote to point back to the original URL later.

If that were done in a way that used no-copy cloning on file systems that support it (e.g. APFS), that could be really fast since the bare repository typically has few but large files (typically mainly a single pack).